### PR TITLE
[AWS] Add dot_expander processor into metrics ingest pipeline

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## Proposed commit message

When we try to ingest CloudWatch metrics using Firehose, metrics can be rerouted to different data streams based on the routing rules. But then the ingest pipelines in each metrics data stream are not applied for some reason to the routed documents. For example `aws.rds.metrics.CPUUtilization.avg` is not renamed to `aws.rds.cpu.total.pct`. Turned out this is because when documents are constructed, these fields are stored as dotted fields instead of json. Using the `dot_expander` processor will expand these fields with dots into object fields which allows fields with dots to be accessible by other processors later. 

This can be fixed in the code that constructs the Firehose documents but I think we should either way add the dot_expander processor here in the ingest pipeline. With this change, if users ingest their own documents, the dotted fields will be taken care of as well. Also change here in the integrations repo can be released right away 🙂 

This PR also copies `cloud.account.id` field to `cloud.account.name` when the account name is not available. For example when metrics are collected from firehose, only the account ID is available. In this case, we need to copy the ID to the name field in order to be able to use the existing ootb dashboards.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

**Comment by agithomas** [comment]
I think the change may be needed for `aws_bedrock` dataset `runtime`

**Comment by agithomas** [comment]
The general guidelines are available [here](https://docs.google.com/document/d/1QL4v1kJriefT_IitM2Th8b2P60hm5Z2V19F1Vzq6YLE/edit?pli=1).

**Comment by kaiyan-sheng** [comment]
> I think the change may be needed for `aws_bedrock` dataset `runtime`, additionally

@agithomas Thanks for the comment! Since bedrock is not supported in firehose rerouting yet, I just created a separate PR to add into Firehose rerouting for metrics and the dot_expander processor: https://github.com/elastic/integrations/pull/10699

**Review by efd6 — COMMENTED** [review]
ISTM that all the test cases here already have their dots expanded. Can we have tests that follow the code path that is being added?

**Comment by kaiyan-sheng** [comment]
@efd6 Oh yes sorry that was the intention! Thanks for catching it. Just made the change.

**Review by efd6 — APPROVED** [review]
Thanks (as codeowner only).

**Review by mbondyra — APPROVED** [review]
LGTM 👌🏼

File: packages/aws/data_stream/ec2_metrics/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  - script:
+      lang: painless
+      description: This script fills cloud.account.name with cloud.account.id when the name is not available.
+      source: |
+        if(ctx.cloud?.account?.id != null && ctx.cloud?.account?.name == null) {
+            ctx.cloud.account.name = ctx.cloud.account.id;
+        }
+        if(ctx.aws?.ec2?.metrics?.CPUUtilization?.avg != null && ctx.host?.cpu?.usage == null) {
+            ctx.aws.ec2.metrics.CPUUtilization.avg = ctx.aws.ec2.metrics.CPUUtilization.avg / 100;
+        }
```
Comment (zmoog) [inline packages/aws/data_stream/ec2_metrics/elasticsearch/ingest_pipeline/default.yml:17]: It seems we are mixing two different tasks and the description only covers the `cloud.account.{id,name}` thing.

**Review by zmoog — APPROVED** [review]
LGTM overall, I only left a couple of non-blocking comments about the Painless scripts.

File: packages/aws/data_stream/ec2_metrics/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (kaiyan-sheng) [reply packages/aws/data_stream/ec2_metrics/elasticsearch/ingest_pipeline/default.yml]: ahh yes changed the description! Thanks!

**Review by agithomas — APPROVED** [review]
LGTM!

File: packages/aws/data_stream/dynamodb/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+description: "Pipeline for DynamoDB metrics"
+
+processors:
+  - dot_expander:
+      field: "*"
+      ignore_failure: true
+  - set:
+      field: cloud.account.name
```
Comment (agithomas) [inline packages/aws/data_stream/dynamodb/elasticsearch/ingest_pipeline/default.yml:9]: Below is the extract from the sample events and here `cloud.account.name` and `cloud.account.id` are completely different values. 

So, why are we copying it here?

```   
         "cloud": {
                "account": {
                    "id": "627286350134",
                    "name": "MonitoringAccount"
                },
```

Comment (agithomas) [reply packages/aws/data_stream/dynamodb/elasticsearch/ingest_pipeline/default.yml:9]: Are we seeing a situation when we have `cloud.account.id` available but not `cloud.account.name`?

Comment (kaiyan-sheng) [reply packages/aws/data_stream/dynamodb/elasticsearch/ingest_pipeline/default.yml:9]: @agithomas Yes! The account name is only available for metrics from elastic-agent/metricbeat. When metrics are collected from firehose, this field will always be empty which breaks the dashboards. This `copy` should only happen when account name is not available and should not overwrite.

Comment (kaiyan-sheng) [reply packages/aws/data_stream/dynamodb/elasticsearch/ingest_pipeline/default.yml:9]: @agithomas I also added the set processor for cloud.account.name into aws_bedrock as you preferred. Thanks!

**Pull request merged**